src/hotspot/share/interpreter/interpreterRuntime.cpp

```diff
@@ -299,76 +306,71 @@ JRT_ENTRY(void, InterpreterRuntime::uninitialized_static_inline_type_field(JavaT
   }
 JRT_END
 
-JRT_ENTRY(void, InterpreterRuntime::read_flat_field(JavaThread* current, oopDesc* obj, int index, Klass* field_holder))
-  Handle obj_h(THREAD, obj);
-
+JRT_ENTRY(void, InterpreterRuntime::read_flat_field(JavaThread* current, oopDesc* obj, ResolvedFieldEntry* entry))
   assert(oopDesc::is_oop(obj), "Sanity check");
+  Handle obj_h(THREAD, obj);
 
-  assert(field_holder->is_instance_klass(), "Sanity check");
-  InstanceKlass* klass = InstanceKlass::cast(field_holder);
-
-  assert(klass->field_is_flat(index), "Sanity check");
+  InstanceKlass* holder = InstanceKlass::cast(entry->field_holder());
+  assert(entry->field_holder()->field_is_flat(entry->field_index()), "Sanity check");
 
-  InlineKlass* field_vklass = InlineKlass::cast(klass->get_inline_type_field_klass(index));
+  InlineLayoutInfo* layout_info = holder->inline_layout_info_adr(entry->field_index());
+  InlineKlass* field_vklass = layout_info->klass();
```

Comment:
Why `vklass`?  Valhalla Klass?

Reply:
Value Klass. InstanceKlass pointers are often named ik, but now those two letters can be interpreted as either InstanceKlass or InlineKlass. So, I often use vk or vklass to designate a pointer to an InlineKlass.